pkg/manifest/manifest.go

```diff
@@ -38,3 +39,33 @@ func GetInstalledManifestForPackage(ctx context.Context, pkg ctrlpkg.Package) (*
 		return nil, ErrPackageNoManifest
 	}
 }
+
+func GetManifestForPackage(
+	ctx context.Context,
+	pkg ctrlpkg.Package,
+	version string,
+) (*v1alpha1.PackageManifest, error) {
+	repoClient := cliutils.RepositoryClientset(ctx)
+	var packageIndex repo.PackageIndex
+	err := repoClient.ForPackage(pkg).FetchPackageIndex(pkg.GetSpec().PackageInfo.Name, &packageIndex)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching package index: %w", err)
+	}
+
+	for _, v := range packageIndex.Versions {
+		if v.Version == version {
+			var manifest v1alpha1.PackageManifest
+			err := repoClient.ForPackage(pkg).FetchPackageManifest(
+				pkg.GetSpec().PackageInfo.Name,
+				version,
+				&manifest,
+			)
+			if err != nil {
+				return nil, fmt.Errorf("error fetching manifest: %w", err)
+			}
+			return &manifest, nil
+		}
+	}
+
+	return nil, fmt.Errorf("version %s not found for package %s", version, pkg.GetSpec().PackageInfo.Name)
```

Comment:
Please correct me if I'm wrong, but I think this is unnecessarily complex. 
```suggestion
	var manifest v1alpha1.PackageManifest
	err := repoClient.ForPackage(pkg).FetchPackageManifest(
		pkg.GetSpec().PackageInfo.Name,
		version,
		&manifest,
	)
	if err != nil {
		return nil, fmt.Errorf("error fetching manifest: %w", err)
	}
	return &manifest, nil
```